### PR TITLE
feat: (Superadmin)Retrieves emails in waitlist

### PR DESCRIPTION
**Pull request by ericBlack1**
## Description
The PR implements the functionality for getting all emails that were used to signup for the waitlist

## Related Issue (Link to issue ticket)
This project only accepts pull requests related to open issues 
If suggesting a new feature or change, please discuss it in an issue first
If fixing a bug, there should be an issue describing it with steps to reproduce 
Please link to the issue here: [#235](https://github.com/hngprojects/hng_boilerplate_python_fastapi_web/issues/235)

## Motivation and Context
This feature provides a comprehensive view of all users who have signed up, enabling superadmins to monitor and manage the waitlist effectively. It will also ensure that only users with superadmin privileges can access this endpoint, maintaining data security and restricting access to authorized personnel.

## How Has This Been Tested?
The functionality has been tested using pytest with the following test cases:

- [x] test_get_all_waitlist_emails_success: Verifies successful waitlist retrieval
- [x] test_get_all_waitlist_emails_non_superadmin: Verifies for unauthorized access 

## Screenshots (if appropriate - Postman, etc):
Screenshot from 2024-07-31 02-10-42

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: tests/v1/waitlist/test_add_user_waitlist.py [hunk 1]
```diff
@@ -66,3 +87,37 @@ def test_add_user_to_waitlist_duplicate_email(self, mock_service, client):
...
+    def test_get_all_waitlist_emails_success(self, mock_service, client):
+        print("Current Dependency Override for get_super_admin:", app.dependency_overrides.get(get_super_admin))
+        # Mock return value for fetch_all method
+        mock_service.return_value = [
+            MagicMock(email="test@example.com"),
+            MagicMock(email="duplicate@example.com")
+        ]
+
+        response = client.get("/api/v1/waitlists/users")
+
```
Comment (Goketech) [inline tests/v1/waitlist/test_add_user_waitlist.py:112]: write tests in a new file

**Review by Goketech — CHANGES_REQUESTED** [review]

File: tests/v1/waitlist/test_add_user_waitlist.py (hunk 1 above)
Comment (ericBlack1) [reply tests/v1/waitlist/test_add_user_waitlist.py]: It's done

File: api/v1/routes/waitlist.py [hunk 2]
```diff
@@ -95,4 +96,19 @@ def admin_add_user_to_waitlist(
...
+    return JsonResponseDict(**resp)
+
+@waitlist.get("/users", response_model=success_response, status_code=200)
+async def get_all_waitlist_emails(
+    request: Request,
+    db: Session = Depends(get_db),
+    admin=Depends(get_super_admin)
+):
+    waitlist_users = waitlist_service.fetch_all(db)
+    emails = [user.email for user in waitlist_users]
```
Comment (Goketech) [inline api/v1/routes/waitlist.py:108]: return an array of objects containing the email and full_name

**Review by Goketech — CHANGES_REQUESTED** [review]

File: api/v1/routes/waitlist.py (hunk 2 above)
Comment (ericBlack1) [reply api/v1/routes/waitlist.py:108]: I did it this way because the github issue i was working from, it just returned an array only containing the emails

**Review by joboy-dev — APPROVED** [review]

**Comment by joboy-dev** [comment]
Make the changes that @Goketech requested. It would be useful for the frontend

**Comment by ericBlack1** [comment]
@Goketech  I've added the names when displaying the waitlist

**Comment by Goketech** [comment]
pull from dev and make sure your pr is up to date

**Comment by ericBlack1** [comment]
done

File: alembic/versions/28747295af93_add_filter_status_field_to_product_model.py [hunk 3]
```diff
@@ -13,7 +13,7 @@
 
 # revision identifiers, used by Alembic.
 revision: str = '28747295af93'
-down_revision: Union[str, None] = '5f9d01dc6cc9'
+down_revision: Union[str, None] = '5f8e13d19ccb'
 branch_labels: Union[str, Sequence[str], None] = None
```
Comment (Goketech) [inline alembic/versions/28747295af93_add_filter_status_field_to_product_model.py:17]: you updated migrations?
revert this please

File: api/v1/routes/auth.py [hunk 4]
```diff
@@ -228,6 +228,39 @@ async def verify_signin_token(token_schema: TokenRequest, db: Session = Depends(
...
+        value=refresh_token,
+        expires=timedelta(days=30),
+        httponly=True,
+        secure=True,
+        samesite="none",
+    )
+
+    return response
+
+
```
Comment (Goketech) [inline api/v1/routes/auth.py:263]: this is a duplicate you're adding

**Review by Goketech — CHANGES_REQUESTED** [review]

File: alembic/versions/28747295af93_add_filter_status_field_to_product_model.py (hunk 3 above)
Comment (ericBlack1) [reply alembic/versions/28747295af93_add_filter_status_field_to_product_model.py]: Okay now look at what happens when I do

image

Comment (ericBlack1) [reply alembic/versions/28747295af93_add_filter_status_field_to_product_model.py]: image

File: api/v1/routes/auth.py (hunk 4 above)
Comment (ericBlack1) [reply api/v1/routes/auth.py:263]: Those are changes i pulled

Comment (ericBlack1) [reply api/v1/routes/auth.py:263]: image

Comment (ericBlack1) [reply api/v1/routes/auth.py:263]: image

Comment (ericBlack1) [reply api/v1/routes/auth.py:263]: It says there "Joshua Oloton, 2 days ago"

Comment (ericBlack1) [reply api/v1/routes/auth.py:263]: @Goketech  I have not touched that file

**Review by Goketech — APPROVED** [review]

**Review by joboy-dev — APPROVED** [review]

**Pull request merged**